app/services/sso/openid_connect.rb

```diff
@@ -4,27 +4,48 @@ class OpenidConnect < Base
     attr_reader :current_user
 
     def available?
+      if session[:sso_method] == "SSO::OpenidConnect"
+        session.delete(:sso_method)
```

Comment:
should we delete it only if there is no user? 
```suggestion
        session.delete(:sso_method) unless session[:user]
```
or maybe add it into the condition?

Reply:
I was thinking about this one and did not understand the reason for this condition, can you explain? From what I understood, since `available` method is called upon login, we will never have a user in the session, since the previous one must have been deleted through the `reset_session`.

Reply:
It is called indirectly every time `authenticate` is called, which is on every request if i'm not mistaken.